### DISCUSSION TITLE
Disable automated restart procedure

### DISCUSSION THREAD

**Discussion by tomer-ds** [gnzsnz/ib-gateway-docker#164] [Q&A] [answered]
Hi,

Is there a way to completely disable IBC's automated restart of IBGateway?
In communication with IB they mentioned that TWS MUST be restarted every 24 hours, but IBGateway does not.
Since I am using IBGateway I tried to disable the restart by removing the configuration, but there seems to be a default.

Is there some way to disable this functionality?

**Comment by gnzsnz** [comment]
I'm not aware of this possibility this is very interesting.

Do you have any reference from IBKR regarding this affirmation?

Have you been able to set up IBC to work in this way? what is the delta between IBC settings and current `ib-dateway-docker` env vars?

If possible, this looks promising.

**Reply by gnzsnz** [reply, marked as answer]
possibly a duplicate with #145

**Reply by gnzsnz** [reply]
duplicate

**Reply by GonzRon** [reply]
> I'm not aware of this possibility this is very interesting.
> 
> Do you have any reference from IBKR regarding this affirmation?
> 
> Have you been able to set up IBC to work in this way? what is the delta between IBC settings and current `ib-dateway-docker` env vars?
> 
> If possible, this looks promising.

https://www.multicharts.com/discussion/viewtopic.php?t=45212


_> IB'S REPLY
> 
> The IBGateway does not need to be restarted once a day like the Trader Workstation, so it can stay up and running 24 hours over 24 but nevertheless the IBGateway is as well affected by the restart of our servers which happens once a day according to the schedule mentioned in this page:
> [http://www.interactivebrokers.com/en/so ... entity=llc](http://www.interactivebrokers.com/en/software/systemStatus.php?ib_entity=llc)
> 
> During this time frame the systems could be not available for a certain period. This will cause the IBGateway to disconnect from the servers. Usually it will reconnect automatically once the servers are up again.
> 
> Since this is a scheduled maintenance which is performed every day, we do not send any alerts, email or warnings, but since you user looks like connecting from a Linux machine, you might want to create a script which checks the ibgateway.DAY.log (When DAY is the first three letter of the current weekday) which is located in the installation directory of the IBGateway and sends an email (via sendmail) in case the keywords "Disconnection" or "Disconnected" are logged.
>_ 

The maintenance windows:
https://www.interactivebrokers.com/en/software/systemStatus.php?ib_entity=llc

![image](https://github.com/user-attachments/assets/c54a4506-ce27-45c8-b030-c48f34717881)
